File: game_modules/faction_objects/colonies.py
class Colony:
	def __init__(self, colony_id, name, planet_instance, empire_instance, galaxy, **kwargs):
		# Colony Location and General information
		self.galaxy = galaxy
		self.name = name

		# Parents
		self.parent_empire = empire_instance
		self.parent_planet = planet_instance
		self.ids = {
			'self': colony_id,

			'star': planet_instance.ids['star'],
			'planet': planet_instance.ids['self'],

			'empire': empire_instance.ids['self']
		}

		# Colony Type information
		self.colony_type = 'mixed'  # Military, Mixed, Civilian

		# Colony buildings information
		self.buildings = {
			# Quantity buildings (Can keep expanding more facilities)
			'mines': 100,
			'factories': 100,
		}
		self.installations = {
			# The are buildings that have instances attached with special and unique stats for each building.
			# 'name': id
		}

		self.construction_projects = {}  # A dict of {id's: instance} of all construction projects assigned to the colony.

		# Colony Storage
		self.resource_storage = {
			'water': 999999999,
			'building_materials': 999999999999
		}

		self.__dict__.update(kwargs)

		empire_instance.colonies[self.ids['self']] = self
		planet_instance.colonies[self.ids['self']] = self

	def update_construction(self, galaxy, empire):
		for construction_project in self.construction_projects:
			galaxy.construction_projects[construction_project].tick_construction(empire, self)

	# SETTERS
	def add_buildings(self, building):
		if type(building) == str:
			self.buildings[building] += 1
		else:
			self.galaxy.add_objects({'installations': building})
			self.installations[building.name] = building.id

	def serialize(self):
		dictionary = dict(self.__dict__)
		dictionary['construction_projects'] = dict()
		dictionary['installations'] = dict()
		del dictionary['galaxy']
		return dictionary


class Installation:
	def __init__(self, name, installation_id, galaxy):
		# General information and identification of the installation.
		self.galaxy = galaxy
		self.name = name
		self.id = installation_id

		self.stats = {}

	def serialize(self):
		dictionary = dict(self.__dict__)
		del dictionary['galaxy']
		del dictionary['parent_empire']
		del dictionary['parent_planet']
		return dictionary

File: static/js/classes/faction_objects/colonies.js
/**
 * Created by Tomca on 9/03/2017.
 */

function Colony_object (name, ids, colony_type, buildings, installations, construction_projects, resource_storage, world) {
    this.world = world;

    this.name = name;
    this.ids = ids;

    this.colony_type = colony_type;

    this.buildings = buildings;
    this.installations = installations;

    this.construction_projects = construction_projects;

    this.resource_storage = resource_storage;

    var config = this.world.get_config();

    var parent_empire = config.empires[this.ids['empire']];
    var parent_planet = config.stars[this.ids['star']].planets[this.ids['planet']];
    parent_empire.colonies[this.ids['self']] = this;
    parent_planet.colonies[this.ids['self']] = this;

    this.parent_empire = parent_empire;
    this.parent_planet = parent_planet;
}

Colony_object.prototype.delete_instance = function () {
    delete this.parent_empire.colonies[this.ids['self']];
    delete this.parent_planet.colonies[this.ids['self']];

    for (var installation_id in this.installations) {
        delete this.installations[installation_id].parent_colony;
    }

    for (var construction_project in this.construction_projects) {
        delete this.construction_projects[construction_project].parent_colony;
    }
};

function _create_colony(colony_instance, world) {
    new Colony_object(
        colony_instance['name'],
        colony_instance.ids,
        colony_instance.colony_type,
        colony_instance.buildings,
        {},
        {},
        colony_instance.resource_storage,
        world
    )
}
